vendor/views/config.py

```diff
@@ -0,0 +1,280 @@
+from django.contrib.sites.models import Site
+from django.views.generic.list import ListView
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+from django.views.generic.edit import FormView, UpdateView
+from django.shortcuts import redirect, render
+
+from vendor.views.mixin import SiteOnRequestFilterMixin, get_site_from_request
+from vendor.config import PaymentProcessorSiteConfig, PaymentProcessorForm,\
+    StripeConnectAccountConfig, StripeConnectAccountForm,\
+    VendorSiteCommissionForm, VendorSiteCommissionConfig
+from siteconfigs.models import SiteConfigModel
+
+
+class PaymentProcessorSiteConfigsListView(ListView):
+    template_name = 'vendor/manage/config_list.html'
+    model = SiteConfigModel
+
+    def get_queryset(self):
+        payment_processor = PaymentProcessorSiteConfig()
+
+        return SiteConfigModel.objects.filter(key=payment_processor.key)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        processot_config = PaymentProcessorSiteConfig()
+
+        context['title'] = 'Payment Processors Configured'
+        context['config_key'] = processot_config.key
+        context['new_url'] = reverse('vendor_admin:manager-config-processor-create')
+
+        return context
+
+
+class PaymentProcessorCreateConfigView(FormView):
+    template_name = 'vendor/manage/config.html'
+    form_class = PaymentProcessorForm
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(**kwargs)
+        processor_config = PaymentProcessorSiteConfig()
+        form = self.get_form()
+
+        existing_stripe_configs = SiteConfigModel.objects.filter(key=processor_config.key)
+        sites = Site.objects.exclude(pk__in=[config.site.pk for config in existing_stripe_configs])
+        form.fields['site'].queryset = sites
+
+        context['title'] = _("Payment Processor Config")
+        context['form'] = form
+        context['cancel_url'] = reverse('vendor_admin:manager-config-processor-list')
+
+        return context
+
+    def form_valid(self, form):
+        site = form.cleaned_data['site']
+
+        processor_config = PaymentProcessorSiteConfig(site)
+        processor_config.save(form.cleaned_data["payment_processor"], "payment_processor")
+        
+        return redirect('vendor_admin:manager-config-processor-list')
+
+
+class PaymentProcessorUpdateConfigView(UpdateView):
+    template_name = 'vendor/manage/config.html'
+    model = SiteConfigModel
+    form_class = PaymentProcessorForm
+
+    def get_form(self, **kwargs):
```

Comment:
i don't think this method needs to be explicitly defined here; doesn't seem like you're altering the form in any way and you have the class declared above

Reply:
kwargs removed

Reply:
@rhimmelbauer i dont have any issue with the kwargs argument; what i actually meant was that i don't think you need to have the get_form method here, since you're already declaring the form_class above

Reply:
I see what you mean duh. thanks

Reply:
I remember now. I am over righting it because I want it to ignore built in functionality it has whit the UpdateView Class